AWSClientRuntime/Sources/Errors/UnknownAWSHttpServiceError.swift

```diff
@@ -23,10 +25,21 @@ public struct UnknownAWSHttpServiceError: AWSHttpServiceError, Equatable {
 }
 
 extension UnknownAWSHttpServiceError {
-    public init(httpResponse: HttpResponse, message: String? = nil, requestID: String? = nil) {
+    public init(
+        httpResponse: HttpResponse,
+        message: String? = nil,
+        requestID: String? = nil,
+        errorCode: String? = nil
+    ) {
```

Comment:
non-blocking, but would you mind adding a doc comment for this initializer while you're here?

Reply:
Added